compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -4531,12 +4538,47 @@ abstract class GenJSCode extends plugins.PluginComponent
     sym.isAnonymousClass && AllJSFunctionClasses.exists(sym isSubClass _)
 
   private def isRawJSCtorDefaultParam(sym: Symbol) = {
+    isCtorDefaultParam(sym) &&
+    isRawJSType(patchedLinkedClassOfClass(sym.owner).tpe) &&
+    nme.defaultGetterToMethod(sym.name) == nme.CONSTRUCTOR
+  }
+
+  private def isJSNativeCtorDefaultParam(sym: Symbol) = {
+    lazy val companionClass = patchedLinkedClassOfClass(sym.owner)
+    isCtorDefaultParam(sym) &&
+    isRawJSType(companionClass.tpe) &&
+    !isScalaJSDefinedJSClass(companionClass)
+  }
+
+  private def isCtorDefaultParam(sym: Symbol) = {
     sym.hasFlag(reflect.internal.Flags.DEFAULTPARAM) &&
     sym.owner.isModuleClass &&
-    isRawJSType(patchedLinkedClassOfClass(sym.owner).tpe) &&
     nme.defaultGetterToMethod(sym.name) == nme.CONSTRUCTOR
   }
 
+  private def hasDefaultCtorArgsAndRawJSModule(classSym: Symbol): Boolean = {
+    /* Get the companion module class.
+     * For inner classes the sym.owner.companionModule can be broken,
+     * therefore companionModule is fetched at uncurryPhase.
+     */
+    val companionModule = enteringPhase(currentRun.uncurryPhase) {
+      classSym.companionModule
```

Comment:
I had to access the companion module the exact same way in `GenJSExports` it might be a good idea to factor it out. But I'm not sure where would be the most appropriate place to place it.